clippy_lints/src/indexing_slicing.rs

```diff
@@ -42,12 +42,15 @@ declare_clippy_lint! {
 
 declare_clippy_lint! {
     /// ### What it does
-    /// Checks for usage of indexing or slicing. Arrays are special cases, this lint
-    /// does report on arrays if we can tell that slicing operations are in bounds and does not
-    /// lint on constant `usize` indexing on arrays because that is handled by rustc's `const_err` lint.
+    /// Checks for usage of indexing or slicing that may panic at runtime.
+    ///
+    /// This lint does does not report on indexing or slicing
```

Comment:
typo: twice does

Reply:
It seems like previous statement had more details?

Reply:
@alex-semenyuk unfortunately the previous statement was wrong! 

> this lint does report on arrays if we can tell that slicing operations are in bounds

This isn't correct, the lint only fires when the slicing is possibly out of bounds.

```
#![warn(clippy::indexing_slicing)]

fn main() {
    let y = [0, 1, 2, 3];

    let i = 10;
    let j = 20;
    let _ = &y[1..2]; // no lint, definitely in bounds
    let _ = &y[i..j]; // lint, possibly out of bounds
}
```

> that is handled by rustc's `const_err` lint.

It's actually `unconditional_panic` in rustc. I don't think it's terribly relevant here, as this lint is concerned with code that can panic, depending on the runtime value.